website/content/docs/install/cloud-auto-join.mdx

```diff
@@ -138,12 +142,20 @@ $ consul agent -retry-join "provider=aws tag_key=... tag_value=..."
 - ECS task role metadata (container-specific).
 - EC2 instance role metadata.
 
-The only required IAM permission is `ec2:DescribeInstances`, and it is
+To discover EC2 consul-servers, the only required IAM permission is `ec2:DescribeInstances`, and it is
 recommended that you make a dedicated key used only to auto-join the datacenter. If the
 region is omitted it will be discovered through the local instance's [EC2
 metadata
 endpoint](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html).
 
+To discover ECS consul-servers, the following IAM permissions are required on the AWS ECS Task Role associated with the Service performing discovery.
+- `ecs:ListClusters` (only used when `ecs_cluster` is not provided)
+- `ecs:ListServices` (only used when `ecs_cluster` is not provided)
+- `ecs:DescribeServices` (only used when `ecs_cluster` is not provided)
+- `ecs:ListTasks`
+- `ecs:DescribeTasks`
+If the region is omitted it will be discovered through the local instance's [ECS V4 metadata endpoint](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html).
+
```

Comment:
```suggestion
The AWS ECS task role associated with the service attempting to discover the `consul-server` must have the following IAM permissions:
- `ecs:ListClusters` (only used when `ecs_cluster` is not provided)
- `ecs:ListServices` (only used when `ecs_cluster` is not provided)
- `ecs:DescribeServices` (only used when `ecs_cluster` is not provided)
- `ecs:ListTasks`
- `ecs:DescribeTasks`
If the region is omitted from the configuration, Consul obtains it from the local instance's [ECS V4 metadata endpoint](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html).

```
Is that introduction any clearer? Ignore if it doesn't seem like an improvement.